yt/frontends/amrvac/data_structures.py

```diff
@@ -171,16 +171,33 @@ def __init__(self, filename, dataset_type='amrvac',
                                             units_override=units_override, unit_system=unit_system)
 
         self._parfiles = parfiles
-        if parfiles is not None:
-            self.namelist = read_amrvac_namelist(parfiles)
-            if "hd_list" in self.namelist:
-                namelist_gamma = self.namelist["hd_list"].get("hd_gamma", None)
-                if namelist_gamma is not None and not self.gamma == namelist_gamma:
-                    mylog.error("Inconsistent values in gamma: datfile {}, parfiles {}".format(self.gamma, namelist_gamma))
 
-        else:
-            # todo: add a warning here when having this parameter becomes useful
-            self.namelist = None
+        namelist = None
+        namelist_gamma = None
+        c_adiab = None
+        e_is_internal = None
+        if parfiles is not None:
```

Comment:
Do you think there is ever a chance that `parfiles` will be an iterable that is not a list/tuple?  i.e., if it's a string, will this cause failure?  If so, you could wrap in `yt.funcs.ensure_list` to be on the safe side.

Reply:
It's actually _meant_ to be a string in 99% of relevant cases, because few people in the AMRVAC community know and use the multiple parfile approach to set up their simulations. In any case, the one time that the `parfile` argument is used is the line right following this extract, namely `namelist = read_amrvac_namelist(parfiles)`, and this is tested against a string path in `tests/test_read_amrvac_namelist.py`.